crates/ruff/src/rules/ruff/rules/mutable_defaults_in_class_fields.rs

```diff
@@ -165,8 +195,13 @@ fn is_class_var_annotation(context: &Context, annotation: &Expr) -> bool {
     context.match_typing_expr(value, "ClassVar")
 }
 
-/// RUF009
-pub fn function_call_in_dataclass_defaults(checker: &mut Checker, body: &[Stmt]) {
+/// RUF009/RUF011
+pub fn function_call_in_class_defaults(
+    checker: &mut Checker,
+    body: &[Stmt],
+    is_dataclass: bool,
+    emit_dataclass_error: bool,
```

Comment:
Nit: Do we need both booleans. It seems that `emit_dataclass_error` is always `true` if `is_dataclass` is true

Reply:
Well, it depends on if you think RUF008/9 are distinct from RUF010/11. Should the latter issue the same error as the former if the former are not enabled? Probably not right?

I would personally rather just collapse the error but I wasn't sure if it was acceptable to break backwards compatibility like that. That said, RUF008/9 are quite new so maybe it's okay?

Reply:
Oh I see, the nesting in the call-site is different than I thought. But does that mean that Ruff now emits two errors when seeing a non-immutable function call: Once the dataclass error and once the normal error if both rules are enabled? 

@charliermarsh what's your opinion on merging the rules and/or parametrizing the diagnostics with whether it is a dataclass or not?

Reply:
How about this: I'll put up a PR that expands RUF008 to all classes, not just dataclasses. I think that's uncontroversial, since mutable defaults are never a good idea. There are some issues with RUF009 because we need exemptions for `field_specifiers` in general for `@dataclass_transforms`, which is not easy to do in general. Then we can sort out whether we want to keep RUF009 scoped to dataclasses or expand it to any class with exemptions for dataclass-like things.

Reply:
Sounds good to me. @charliermarsh what's your take?

Reply:
I've added my opinion over in https://github.com/charliermarsh/ruff/pull/4390#issuecomment-1546348728